src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -132,22 +164,59 @@ public ObservableList<Customer> getFilteredCustomerList() {
     public void updateFilteredCustomerList(Predicate<Customer> predicate) {
         requireNonNull(predicate);
         filteredCustomers.setPredicate(predicate);
-        if (filteredCustomers.size() > 0) {
-            selectCustomer(filteredCustomers.get(0));
-        }
     }
 
+    //=========== Filtered Commission List Accessors =============================================================
+
+    /**
+     * Returns an unmodifiable view of the list of {@code Commission} backed by the internal list of
+     * {@code versionedAddressBook}
+     */
+    @Override
+    public ObservableList<Commission> getFilteredCommissionList() {
+        return filteredCommissions;
+    }
+
+    @Override
+    public void updateFilteredCommissionList(Predicate<Commission> predicate) {
+        requireNonNull(predicate);
+        filteredCommissions.setPredicate(predicate);
+    }
+
+    private Predicate<Commission> getActiveCustomerCommissionsPredicate() {
+        return commission -> commission.getCustomer().isSameCustomer(getSelectedCustomer().getValue());
+    }
+
+    public void updateFilteredCommissionListToActiveCustomer() {
+        updateFilteredCommissionList(getActiveCustomerCommissionsPredicate());
+    }
+
+    //=========== Selected Customer =============================================================
+
 
     @Override
     public void selectCustomer(Customer customer) {
         this.selectedCustomer.setValue(customer);
+        updateFilteredCommissionListToActiveCustomer();
     }
 
     @Override
     public ObservableObject<Customer> getSelectedCustomer() {
         return this.selectedCustomer;
     }
 
+    /**
+     * Return whether there is an active customer.
```

Comment:
If I'm not mistaken, the first word of the JavaDoc for a method should be plural?

```suggestion
     * Returns whether there is an active customer.
```